conformance/05_referrers.go

```diff
@@ -0,0 +1,276 @@
+			})
+		})
+
+		g.Context("Error codes", func() {
+			g.Specify("400 response body should contain OCI-conforming JSON message", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.PUT, "/v2/<name>/manifests/<reference>",
+					reggie.WithReference("sha256:totallywrong")).
+					SetHeader("Content-Type", "application/vnd.oci.image.manifest.v1+json").
+					SetBody(invalidManifestContent)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAny(
+					Equal(http.StatusBadRequest),
+					Equal(http.StatusNotFound)))
+				if resp.StatusCode() == http.StatusBadRequest {
+					errorResponses, err := resp.Errors()
+					Expect(err).To(BeNil())
+
+					Expect(errorResponses).ToNot(BeEmpty())
+					Expect(errorCodes).To(ContainElement(errorResponses[0].Code))
+				}
+			})
+		})
+
+		g.Context("Teardown", func() {
+			if deleteManifestBeforeBlobs {
+				g.Specify("Delete manifest created in setup", func() {
+					SkipIfDisabled(referrers)
+					RunOnlyIf(runReferencesSetup)
+					req := client.NewRequest(reggie.DELETE, "/v2/<name>/manifests/<digest>", reggie.WithDigest(manifests[4].Digest))
+					resp, err := client.Do(req)
+					Expect(err).To(BeNil())
+					Expect(resp.StatusCode()).To(SatisfyAny(
+						SatisfyAll(
+							BeNumerically(">=", 200),
+							BeNumerically("<", 300),
+						),
+						Equal(http.StatusMethodNotAllowed),
+					))
+				})
+			}
+
+			g.Specify("Delete config blob created in setup", func() {
+				SkipIfDisabled(referrers)
+				RunOnlyIf(runReferencesSetup)
+				req := client.NewRequest(reggie.DELETE, "/v2/<name>/blobs/<digest>", reggie.WithDigest(configs[4].Digest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAny(
+					SatisfyAll(
+						BeNumerically(">=", 200),
+						BeNumerically("<", 300),
+					),
+					Equal(http.StatusMethodNotAllowed),
+				))
+			})
+
+			g.Specify("Delete layer blob created in setup", func() {
+				SkipIfDisabled(referrers)
+				RunOnlyIf(runReferencesSetup)
+				req := client.NewRequest(reggie.DELETE, "/v2/<name>/blobs/<digest>", reggie.WithDigest(layerBlobDigest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAny(
+					SatisfyAll(
+						BeNumerically(">=", 200),
+						BeNumerically("<", 300),
+					),
+					Equal(http.StatusMethodNotAllowed),
+				))
+			})
+
+			if !deleteManifestBeforeBlobs {
+				g.Specify("Delete manifest created in setup", func() {
+					SkipIfDisabled(referrers)
+					RunOnlyIf(runReferencesSetup)
+					req := client.NewRequest(reggie.DELETE, "/v2/<name>/manifests/<digest>", reggie.WithDigest(manifests[4].Digest))
+					resp, err := client.Do(req)
+					Expect(err).To(BeNil())
+					Expect(resp.StatusCode()).To(SatisfyAny(
+						SatisfyAll(
+							BeNumerically(">=", 200),
+							BeNumerically("<", 300),
+						),
+						Equal(http.StatusMethodNotAllowed),
+					))
+				})
```

Comment:
Same requirement to delete the referrer digests here.

Reply:
Fixed.